### ISSUE TITLE
Decide on whether to merge 'document' and 'entity'

### ISSUE THREAD

**Issue by danicatalan** [huridocs/uwazi#821]
_"As a user, I want to create multiple entities on My files and edit their metadata at once."_

- - - - -

**Update**

- [ ] Rethink "Create entity" button.
- [ ] We can use an UI similar to "Create dictionaries" on Settings.
- [ ] As @RafaPolit says, maybe combine the idea of Documents and Entity.

- - - - -

**Update 2**

This story has evolved into the next story:

_"As a user, I want to create entities and documents as the same thing in the system."_

We need to discuss which is the better way to merge both things into one - if we thing that merge it's the best solution.

**Comment by danicatalan** [comment]
cc/ @whyfrycek

**Comment by RafaPolit** [comment]
I still believe the best approach is to combine the idea of Document and Entity.

Users find it confusing, and, quite honestly, the only difference between the two is the presence of a PDF document. We could create always 'entities'. Some will have PDFs, others will have only metadata. The PDF could part of the metadata and it's just another field you add.

We need to simplify this a lot.

**Comment by danicatalan** [comment]
Good point @RafaPolit, we need to discuss this approach as soon as we move this issue into sprint.

**Comment by txau** [comment]
@RafaPolit I agree. We should merge them into "Templates". Then we need to find a way of picking up the kind of view and which document will be displayed in the document virew. Remember that:

- Both documents and entities can have attachments.
- One of the attachments can be selected to display the actual text.

So essentially if a Template has a PDF attached, is marked for viewing and the document view is selected for that template, it would be the equivalent to our current documents.

**Comment by kjantin** [comment]
I can see why merging 'documents' and 'entities' would make things more simple, to some degree. But I want to play devil's advocate, while admitting I don't know much about how this stuff works on the back end.

1. Uwazi was developed as a way to manage large document collections. If we get rid of "documents", are we simply a CMS with some good document management features? 
2. I wonder how this discussion is related to our discussions on improving the way we build connections between documents/entities. I came across the term "join tables" in [this article](https://shomeya.com/articles/knowing-when-to-use-content-types-and-when-to-use-custom-entities) on how to choose between creating a 'content type + fields' or a 'custom entities' in Drupal. Excerpt:

> One of the other scenarios that also lends to using a join table is when the relationship between two entities is more complicated than simple, 'belongs to' or 'has one.' Imagine a relationship between an organization and a list of staff, where the staff have different roles. This is easy, until you get to the condition that staff can have different roles in different organization. Suddenly the role is tied directly to the relationship between the two entities. This is another situation where a join table allows you to specify not only the organization ID and staff ID, but also the role of that relationship.

This makes me wonder if the reason we developed "entities" in the first place was to create the same functionality as this "join table" - something that is simply used to connect many documents to many documents. If this is the case, if all entities are used as a connector, can we simply rename it as "Connection item" or something similar so its purpose is more clear, and then define how this is clearly different from a document? 

And to take it a step further, we could include video and audio into the mix by using their transcripts as the main 'document' and adding other files (video, audio) to that 'document'. (Not sure how to handle images though)
